schema.development.kf

```diff
@@ -637,70 +618,107 @@ action share_credential_through_dag (
+        $issuer_auth_public_key,
+        $original_public_notes,
+        $original_public_notes_signature,
+        $original_content,
+        $original_broader_signature
+    );
+    SELECT CASE WHEN $original_result != 'ok' THEN
+        error('assert_original_credential_signatures:' || $original_result)
     END;
 
-    $result = idos.assert_credential_signatures($issuer_auth_public_key, $public_notes, $public_notes_signature, $content, $broader_signature);
-    SELECT CASE WHEN $result != 'ok' THEN error('assert_credential_signatures:' || $result) END;
+    $copy_result = idos.assert_credential_signatures(
+        $issuer_auth_public_key,
+        '',
+        $copy_public_notes_signature,
+        $copy_content,
+        $copy_broader_signature
+    );
+    SELECT CASE WHEN $copy_result != 'ok' THEN
+        error('assert_copy_credential_signatures:' || $copy_result)
+    END;
 
-    $verifiable_credential_id = idos.get_verifiable_credential_id($public_notes);
+    // Insert original credential
+    $verifiable_credential_id = idos.get_verifiable_credential_id($original_public_notes);
 
     INSERT INTO credentials (id, user_id, verifiable_credential_id, public_notes, content, encryptor_public_key, issuer_auth_public_key, inserter)
     VALUES (
-        $id,
-        $user_id,
+        $original_credential_id,
+        (SELECT DISTINCT user_id FROM wallets WHERE (wallet_type = 'EVM' AND address=$dwg_owner COLLATE NOCASE)
+            OR (wallet_type = 'NEAR' AND public_key = $dwg_owner)),
         CASE WHEN $verifiable_credential_id = '' THEN NULL ELSE $verifiable_credential_id END,
-        $public_notes,
-        $content,
+        $original_public_notes,
+        $original_content,
         $encryptor_public_key,
         $issuer_auth_public_key,
-        get_write_grant_id($user_id)::text
+        $dwg_id::text
     );
 
-    INSERT INTO shared_credentials (original_id, copy_id) VALUES ($original_credential_id, $id);
+    // Insert copy credential
+    INSERT INTO credentials (id, user_id, verifiable_credential_id, public_notes, content, encryptor_public_key, issuer_auth_public_key, inserter)
+    VALUES (
+        $copy_credential_id,
+        (SELECT DISTINCT user_id FROM wallets WHERE (wallet_type = 'EVM' AND address=$dwg_owner COLLATE NOCASE)
+            OR (wallet_type = 'NEAR' AND public_key = $dwg_owner)),
+        NULL,
+        '',
+        $copy_content,
+        $encryptor_public_key,
+        $issuer_auth_public_key,
+        $dwg_id::text
+    );
+
+    INSERT INTO shared_credentials (original_id, copy_id) VALUES ($original_credential_id, $copy_credential_id);
 
     // SELECT is just a strange way of calling a procedure from an action
     SELECT create_access_grant(
-        $grantee_wallet_identifier,
-        $id,
-        $locked_until::int,
-        $content_hash,
-        'write_grant',
-        get_write_grant_id($user_id)::text
+        @caller,
+        $copy_credential_id,
+        0,
```

Comment:
Timelock should be part of the message, if I recall correctly.

Reply:
I was guided by [this](https://www.notion.so/idos-association/Write-Grant-per-atomic-Issuance-15788c585fe080a1833edf8e39505492?pvs=4#15788c585fe080638726ee77e93c9f98)
`ONE Access Grant for the Issuer that has NO Timelock.`
So is this needed to ask `timelock` in the message and then not to use?

Reply:
I'm pretty sure that was a decision born out of trying to make things simple. Since adding the timelock to our current implementation is trivial, I argue that we should add it.

Let's disambiguate with Martin.

Reply:
https://idosgroup.slack.com/archives/C07J0SDAW5U/p1738742759795989